src/daemon/ca/manager.rs

```diff
@@ -811,54 +811,70 @@ impl CaManager {
-                                    .set_child_suspended(&ca_handle, &child)
-                                    .await
-                                {
-                                    error!(
-                                        "Could not update status to 'suspended' for inactive child, error: {}",
-                                        e
-                                    );
-                                }
+            join_all(updates).await;
+        }
+    }
 
-                                let req = UpdateChildRequest::suspend();
-                                if let Err(e) = self.ca_child_update(&ca_handle, child, req, actor).await {
-                                    error!("Could not suspend inactive child, error: {}", e);
-                                }
-                            }
+    /// Refresh a single CA with its parents, and possibly suspend inactive children.
+    pub async fn cas_refresh_single(&self, ca_handle: Handle, started: Timestamp, actor: &Actor) {
+        let mut updates = vec![];
+
+        // Set threshold hours if it was configured AND this server has been started
+        // longer ago than the hours specified. Otherwise we risk that *all* children
+        // without prior recorded status are suspended on upgrade, or that *all* children
+        // are suspended if the server had been down for more than the threshold hours.
+        let threshold_seconds = self
+            .config
+            .suspend_child_after_inactive_seconds()
+            .filter(|secs| started < Timestamp::now_minus_seconds(*secs));
+
+        if let Ok(ca) = self.get_ca(&ca_handle).await {
+            for parent in ca.parents() {
+                updates.push(self.ca_sync_parent_infallible(ca_handle.clone(), parent.clone(), actor.clone()));
+            }
+
+            if let Some(threshold_seconds) = threshold_seconds {
+                if let Ok(ca_status) = self.get_ca_status(&ca_handle).await {
+                    let connections = ca_status.get_children_connection_stats();
+
+                    for child in connections.suspension_candidates(threshold_seconds) {
+                        let threshold_string = if threshold_seconds >= 3600 {
+                            format!("{} hours", threshold_seconds / 3600)
+                        } else {
+                            format!("{} seconds", threshold_seconds)
+                        };
+
+                        info!(
+                            "Child '{}' under CA '{}' was inactive for more than {}. Will suspend it.",
+                            child, ca_handle, threshold_string
+                        );
+                        if let Err(e) = self
+                            .status_store
+                            .lock()
+                            .await
+                            .set_child_suspended(&ca_handle, &child)
+                            .await
+                        {
+                            error!(
```

Comment:
Under what circumstances could this fail? Is logging an appropriate response? Not being able to update the status seems quite a severe problem but I haven't dived into the called code tree to see what it could really mean.

Reply:
fair point.. it can't really fail - unless there is an I/O error updating the disk, in which case a panic and a crash is probably the only sensible thing to do

Reply:
pushed a change to panic here - see also issue #675